src/components/ExtensionAccountCheck.tsx

```diff
@@ -16,22 +16,33 @@
 import React from 'react';
 import { Alert } from '.';
 import { useKeyringContext } from '../contexts/KeyringContextProvider';
+import useLoadingApi from '../hooks/useLoadingApi';
 
 interface Props {
   component: JSX.Element;
 }
 
+// TODO: Move this to a more generic error-show component
+const statusFunc = (from: string, state: boolean) => `${from} chain status: ${!state ? 'disconnected' : 'connected'}`;
+
 const ExtensionAccountCheck = ({ component }: Props): JSX.Element => {
   const { extensionExists, accountExists } = useKeyringContext();
+  // TODO: Move this to a more generic error-show component
+  const { sourceReady, targetReady } = useLoadingApi();
```

Comment:
This values are already available on `useSourceTarget`, but i'm fine if you want to retrieve them from the `useLoadingApi` for readability

Reply:
Its not only a matter of readability but also of simplicity.
useSourceTarget carries much more than needed while useLoadingApi hook is straightforward.
For the first you need minimum to do:
```
const {
    sourceChainDetails: {
      sourceApiConnection: { isApiReady: sourceIsApiReady }
    },
    targetChainDetails: {
      targetApiConnection: {  isApiReady: targetIsApiReady }
    }
  } = useSourceTarget();
  ```
  while for the latter its less code and simpler.